src/server/controllers/changePassword.ts

```diff
@@ -77,6 +81,25 @@ export const setPasswordController = (
         });
       }
 
+      // When a jobs user is registering, we'd like to add them to the GRS group.
+      // We only do this for users going through the welcome flow.
+      //
+      // Once they belong to this group, they aren't shown a confirmation page when-
+      // they first visit the jobs site.
+      //
+      // If the SC_GU_U cookie exists, we try to add the user to the group.
+      // If the cookie doesn't exist for some reason, we log the incident.
+      if (clientId === 'jobs' && path === '/welcome') {
+        const SC_GU_U = cookies?.values.find(({ key }) => key === 'SC_GU_U');
+        if (SC_GU_U) {
+          addToGroup(GroupCode.GRS, req.ip, SC_GU_U.value);
```

Comment:
do we need to `await` here ? Could the request lifecycle complete before the API call to Identity is complete ?